src/compiler/scala/tools/nsc/transform/OverridingPairs.scala

```diff
@@ -53,6 +53,19 @@ abstract class OverridingPairs extends SymbolPairs {
       && !exclude(low)                 // this admits private, as one can't have a private member that matches a less-private member.
       && (lowMemberType matches (self memberType high))
     ) // TODO we don't call exclude(high), should we?
+
+    override protected def skipOwnerPair(lowClass: Symbol, highClass: Symbol): Boolean = {
+      // Two Java-defined methods can be skipped in most cases, as javac will check the overrides; skipping is
+      // actually necessary to avoid false errors, as Java doesn't have the Scala's linearization rules. However, when
+      // a Java interface is mixed into a Scala class, mixed-in default methods need to go through override checking.
+      // See scala/bug#12394.
+      lowClass.isJavaDefined && highClass.isJavaDefined && {
+        !lowClass.isJavaInterface && !highClass.isJavaInterface || {
+          val interfaceMixins = base.info.parents.collect { case i if i.typeSymbol.isJavaInterface => i.typeSymbol }
```

Comment:
Meh, this is not enough, a Scala trait mixin that extends a Java interface needs to get the same check